pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -921,7 +929,13 @@ func GetSelectorRegexMap(l7 policy.L7DataMap) (CachedSelectorREEntry, error) {
 			}
 		}
 		mp := strings.Join(reStrings, "|")
+		if rei, ok := mapping[mp]; ok {
+			newRE[selector] = rei
+			// metric here
+			continue
+		}
 		rei, err := re.CompileRegex(mp)
+
```

Comment:
Also added this for the testing to improve it even more;

```suggestion
		mapping[mp] = rei
```